engine/apps/api/views/schedule.py

```diff
@@ -280,6 +282,17 @@ def related_escalation_chains(self, request, pk):
         result = [{"name": e.name, "pk": e.public_primary_key} for e in escalation_chains]
         return Response(result, status=status.HTTP_200_OK)
 
+    @action(detail=True, methods=["get"])
+    def quality(self, request, pk):
+        schedule = self.original_get_object()
+        user_tz, date = self.get_request_timezone()
+        days = int(self.request.query_params.get("days", 90))  # todo: check if days could be calculated more precisely
+
+        events = schedule.filter_events(user_tz, date, days=days, with_empty=True, with_gap=True)
+
+        schedule_score = get_schedule_score(events, days)
```

Comment:
Wondering if this could be a method of the base schedule model, and do the call to filter_events there (so you would only call here to `schedule.get_score(days)`)